### ISSUE TITLE
Python class diagram not created for the base class while works for the derived class

### ISSUE THREAD

**Issue by pprebeg** [doxygen/doxygen#7009]
Python class diagram not created for the base class while it is created for the derived class. The derived 

class diagram shows that the base class is not documented, although it is present in the html text documentation. 
If the UML style is used, the base class diagram is shown, but it is still not connected with the derived class diagram

[test.zip](https://github.com/doxygen/doxygen/files/3212992/test.zip)

**Comment by albert-github** [comment]
- which version of doxygen do you use
- please add some more reference to the questing regarding where you are missing some information. The problem with the missing diagram I d see in e.g. `classpy_d3_v_model_1_1_d3_v_model.html` in e.g.`classpy_d3_v_model_simple_1_1_d3_v_model_ctrl_simple.html` I do see a relation between 2 classes.

**Comment by pprebeg** [comment]
Version 1.8.15 win64 binary
Class diagram is not shown for any of the classes in pyD3VModel.py

**Comment by albert-github** [comment]
The missing images was clear but what are you missing in the images shown (with the UML look) and in the images in e.g. the file `classpy_d3_v_model_simple_1_1_d3_v_model_ctrl_simple.html`

**Comment by pprebeg** [comment]
Sorry, I am not at the computer at the moment.

The problem when UML is on is that the diagram at the derived class is still not connected with the base class documentation. The base class is shown as it is not documented.

IThe related issue is that in the class list hierarchy diagram  two versions of the base classes are shown

**Comment by albert-github** [comment]
When back at the computer (doesn't have to be today) show the missing information. in descriptive text with images or drawn into the images together with the names of the image files / the (html) files where they are used.

**Comment by pprebeg** [comment]
In the classpy_d3_v_model_simple_1_1_d3_v_model_ctrl_simple.html the base is class drawn with a box with a gray border that denotes an undocumented struct or class. It should be drawn with a box with a black border that denotes a documented struct or class. Also, the grayed box does not contain link to the base class. It looks like that the derived class does not recognize that the base class is documented.  
As written above, the related issue is that that in the class list hierarchy diagram (inherits.html) two versions of the base classes are shown. One that comes from the base class documentation, while the other comes from the inherited classes documentation.  The problem remains no mater whether Graphwiz (DOT) is enabled or not.

The Python program that uses those two files runs without any issues.

**Comment by albert-github** [comment]
I looked at the image in `classpy_d3_v_model_simple_1_1_d3_v_model_ctrl_simple.html` and it looks like that the `D3VModelCtrl` is not recognized as part of the `pyD3VModel` package. When changing:
```
class D3VModelCtrlSimple(pyD3VModel.D3VModelCtrl):
```
to
```
class D3VModelCtrlSimple(D3VModelCtrl):
```
the relation looks to me to be present.

Also changing:
```
from pyD3VModel import (D3VModel, D3VModelEntity, D3VModelCtrl)
```
into:
```
from pyD3VModel import D3VModel
from pyD3VModel import D3VModelEntity
from pyD3VModel import D3VModelCtrl
```
looks to me to give good results.

So it looks like that currently doxygen does not support the `from pyD3VModel import (D3VModel, D3VModelEntity, D3VModelCtrl)` construct.

Can you confirm that the last change (splitting the `from ... import`) gives the right results (so based on this we can look for a solution in the code)?

**Comment by albert-github** [comment]
It looks to me that also changing:
```
from pyD3VModel import (D3VModel, D3VModelEntity, D3VModelCtrl)
```
to 
```
from pyD3VModel import D3VModel, D3VModelEntity, D3VModelCtrl
```
yields the good results.

Any reference to the `(...)` syntax ?

**Comment by pprebeg** [comment]
Yes, the options without parenthesis works for me too! My python program (3.7) also runs without any issues. Obviously both options are possible in Python 3.7. This is one of my first Python programs so I am not familiar why the both options are possible and is there any difference. Python is quite liberal language ;)

Thanks a lot for the support!
Thanks a lot for the

**Comment by albert-github** [comment]
In the mean time I found a similar construct in a Python 2 file.

Unfortunately Python is quite liberal.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7011

**Comment by pprebeg** [comment]
Thanks!

**Comment by albert-github** [comment]
Reopened as this will be done at the moment of a release.

**Comment by albert-github** [comment]
An alternative implementation compared to the proposed pull request #7011 has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
